.github/workflows/testflinger/scripts/setup.sh

```diff
@@ -50,32 +50,42 @@ setup_classic() (
   sudo apt-get -qqy install nvidia-container-toolkit
 )
 
-setup_core() (
+setup_core22() (
   set -x
   sudo snap install nvidia-core22
   sudo snap install nvidia-assemble --channel 22/stable
 )
 
-setup() {
-  . /etc/os-release
-
-  install_docker
-
-  if [[ $ID == "ubuntu" ]]; then
-    setup_classic
+setup_core24() {
+  set -x
+  # Install kernel components. 
+  sudo snap install pc-kernel+nvidia-550-ko
+  sudo snap install pc-kernel+nvidia-550-user
+  
+  sudo snap install mesa-2404
+}
```

Comment:
I would keep this function into a subshell as well to keep consistent to what we have for `setup_core22()`:

```suggestion
setup_core24() (
  set -x
  # Install kernel components. 
  sudo snap install pc-kernel+nvidia-550-ko
  sudo snap install pc-kernel+nvidia-550-user
  
  sudo snap install mesa-2404
)
```